opm/autodiff/BlackoilWellModel.hpp

```diff
@@ -175,59 +179,60 @@ namespace Opm {
             using ConvergenceReport = typename WellInterface<TypeTag>::ConvergenceReport;
 
             // create the well container
-            static std::vector<WellInterfacePtr > createWellContainer(const Wells* wells,
-                                                                      const std::vector<const Well*>& wells_ecl,
-                                                                      const bool use_multisegment_well,
-                                                                      const int time_step,
-                                                                      const ModelParameters& param);
+            std::vector<WellInterfacePtr > createWellContainer(const int time_step);
 
-            // Well collection is used to enforce the group control
-            WellCollection* well_collection_;
+            WellState well_state_;
+            WellState previous_well_state_;
 
+            const ModelParameters param_;
             bool terminal_output_;
             bool has_solvent_;
             bool has_polymer_;
-            int current_timeIdx_;
-
+            std::vector<int> pvt_region_idx_;
             PhaseUsage phase_usage_;
             std::vector<bool>  active_;
-            const RateConverterType& rate_converter_;
-            const std::vector<int>& pvt_region_idx_;
-
+            size_t global_nc_;
             // the number of the cells in the local grid
-            int number_of_cells_;
+            size_t number_of_cells_;
+            double gravity_;
+            std::vector<double> depth_;
+
+            DynamicListEconLimited dynamic_list_econ_limited_;
+            std::unique_ptr<RateConverterType> rateConverter_;
+            std::unique_ptr<VFPProperties> vfp_properties_;
 
-            long int global_nc_;
+            SimulatorReport last_report_;
 
             // used to better efficiency of calcuation
             mutable BVector scaleAddRes_;
 
-            void updateWellControls(WellState& xw) const;
+            const Wells* wells() const { return wells_manager_->c_wells(); }
+
+            const Schedule& schedule() const
+            { return ebosSimulator_.gridManager().schedule(); }
```

Comment:
Whils this line is ok for now, it is likely that we will change it in the near future, to support external (to the ebos simulator and well model classes) modification of schedules.

Reply:
as far as I can see, that line should be okay even after that because I'm pretty sure that you will want to update the gridManager's schedule in that case?! (In other words: what's the point of a schedule object that the simulation knows nothing about?)

Reply:
Quite possibly true.